use `pip3 install ...rids@main` in the install script

kevindamm-jigsaw opened the pull request:
This makes the RIDS binary install much cleaner than individually copying the .py code into somewhere in $PATH.

Also increments version to reflect the IOC format updates and src-refactor and fixes up the pyproject.toml setuptools-specific config

kevindamm-jigsaw committed:
use `pip3 install ...rids@main` in the install script

This makes the RIDS binary install much cleaner than individually copying
the .py code into somewhere in $PATH.

Also increments version to reflect the IOC format updates and src-refactor

fortuna reviewed (dismissed):
Much cleaner, thanks!

kevindamm-jigsaw commented on install.sh:
```diff
@@ -16,31 +16,20 @@
 
-# Define sysctl .service config to /etc/systemd and start service
+# install RIDS from repo, including its python dependencies
+pip3 install git+git://github.com/Jigsaw-Code/rids.git@main
```
I was sure that change had been `git commit --amend`ed but looks like it didn't get included when I git pushed.  Updated.

kevindamm-jigsaw commented on install.sh:
```diff
@@ -16,31 +16,20 @@
 
```
I've been testing the re-running since December, it should work.  That's why we disable the RIDS service before updating and re-enabling it, and why source was freshly cloned.

With pip it is a little different because pip will reuse the cached repo files and the eggs of dependencies, but it should fetch the new version and install if the branch name changes (though, it looks like it may check the version field of project and only upgrade if that changes).

I've tried to keep it re-entrant, even with this latest change to add a non-root user for running pip.

kevindamm-jigsaw replied:
I haven't tested the latest version in a droplet yet, I've lost access to the one I had.

kevindamm-jigsaw committed:
pip will not install as root, run it as non-root

may involve creating a jigsaw-rids user and running the service from it

the install.sh script has been updated to check for current user and may
create a new user if it doesn't exist

defaults to username 'rids' but can be
set in the env var $RIDS_USER
